litert/tools/npu_numerics_check.cc

```diff
@@ -99,67 +101,79 @@ Expected<Options> GetOptions() {
   return options;
 }
 
-Expected<void> FillInputTensor(TensorBuffer& buffer, float scale) {
-  LITERT_ASSIGN_OR_RETURN(auto type, buffer.TensorType());
-  const auto& layout = type.Layout();
-  size_t total_elements =
-      std::accumulate(layout.Dimensions().begin(), layout.Dimensions().end(), 1,
-                      std::multiplies<size_t>());
-
-  if (type.ElementType() == ElementType::Float16 ||
-      type.ElementType() == ElementType::Float32 ||
-      type.ElementType() == ElementType::BFloat16) {
-    std::vector<float> data(total_elements);
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = std::sin(i * scale);
-    }
-    return buffer.Write<float>(absl::MakeConstSpan(data));
-  } else if (type.ElementType() == ElementType::Int32) {
-    std::vector<int32_t> data(total_elements);
-    unsigned int seed = 7;
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = rand_r(&seed) % 1024 + 1;
-    }
-    return buffer.Write<int32_t>(absl::MakeConstSpan(data));
-  } else if (type.ElementType() == ElementType::Int16) {
-    std::vector<int16_t> data(total_elements);
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = i % 2048;
-    }
-    return buffer.Write<int16_t>(absl::MakeConstSpan(data));
-  } else if (type.ElementType() == ElementType::Int64) {
-    std::vector<int64_t> data(total_elements);
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = i % 2048;
-    }
-    return buffer.Write<int64_t>(absl::MakeConstSpan(data));
-  } else if (type.ElementType() == ElementType::Int8) {
-    std::vector<int8_t> data(total_elements);
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = i % 256 - 128;
-    }
-    return buffer.Write<int8_t>(absl::MakeConstSpan(data));
-  } else if (type.ElementType() == ElementType::UInt8) {
-    std::vector<uint8_t> data(total_elements);
-    for (size_t i = 0; i < total_elements; ++i) {
-      data[i] = i % 256;
+Expected<std::vector<TensorBuffer>> CreateAndFillInputBuffersWithRandomData(
+    const CompiledModel& compiled_model, size_t signature_index) {
+  LITERT_ASSIGN_OR_RETURN(auto input_buffers,
+                          compiled_model.CreateInputBuffers(signature_index));
+  for (auto& buffer : input_buffers) {
+    LITERT_ASSIGN_OR_RETURN(auto type, buffer.TensorType());
+    if (type.ElementType() == ElementType::Float16 ||
+        type.ElementType() == ElementType::Float32 ||
+        type.ElementType() == ElementType::BFloat16) {
+      LITERT_RETURN_IF_ERROR(
+          tensor_utils::FillBufferWithRandomData<float>(buffer));
```

Comment:
We check `type` here and decide the template type of `FillBufferWithRandomData`, and we check template type in `FillBufferWithRandomData`. I think we can directly check `buffer.TensorType()` in `FillBufferWithRandomData`.

```suggestion
    LITERT_RETURN_IF_ERROR(
          tensor_utils::FillBufferWithRandomData(buffer));
```